modules/db/src/main/java/org/mule/module/db/internal/domain/database/DataSourceConfig.java

```diff
@@ -145,7 +145,9 @@ public boolean isDynamic()
 
     private boolean isDynamicAttribute(String attribute)
     {
-        return !StringUtils.isEmpty(attribute) && muleContext.getExpressionManager().isValidExpression(attribute);
+		return !StringUtils.isEmpty(attribute)
+				&& muleContext.getExpressionManager().isExpression(attribute)
+				&& muleContext.getExpressionManager().isValidExpression(attribute);
```

Comment:
which exact cases are you trying to cover with this? If the attribute is an invalid expression, then it should actually fail, so that the user becomes aware of that and fixes it. This change will consider the value as not an expression, will not fail and the connector will not behave as expected. That situation will be much harder for the user to understand and troubleshoot.

Reply:
Hi,
the isDynamicAttribute method is called with 
`expression	"org.postgresql.Driver"`
and so very often (for each SQL) the log contains:
"Expression string is not an expression.  Use isExpression(String) to validate first"

Asking isExpression before will not create InvalidExpressionExceptions.

1. Creating Exceptions which directly are catched to "return false" slows down the system
2. Logfile is polluted with WARN messages.


Additional Infos:
String comes from here:

```
	<db:generic-config name="Heroku_DataSource"
		driverClassName="org.postgresql.Driver"
		doc:name="Heroku DataSource"
		url="#[com.canda.crm.ResolveHeroku.resolveUrl(flowVars)]">
		<db:pooling-profile maxPoolSize="50" minPoolSize="1" />
	</db:generic-config>

```

Here is the stacktrace:

```
Daemon Thread [[crm-system].HTTP_Listener_Configuration.worker.01] (Suspended (exception InvalidExpressionException))	
	DefaultExpressionManager.validateExpression(String) line: 658	
	DefaultExpressionManager.isValidExpression(String) line: 595	
	DataSourceConfig.isDynamicAttribute(String) line: 148	
	DataSourceConfig.resolveAttribute(String, MuleEvent) line: 153	
	DataSourceConfig.resolve(MuleEvent) line: 50	
	DynamicDbConfigResolver.resolve(MuleEvent) line: 63	
	ConfiguredDbConfigResolver.resolve(MuleEvent) line: 30	
	UpdateMessageProcessor(AbstractDbMessageProcessor).process(MuleEvent) line: 68	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	MessageEnricher$EnricherProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	MessageEnricher$EnricherProcessor(AbstractRequestResponseMessageProcessor).processBlocking(MuleEvent) line: 57	
	MessageEnricher$EnricherProcessor.processBlocking(MuleEvent) line: 216	
	MessageEnricher$EnricherProcessor(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	MessageEnricher.process(MuleEvent) line: 73	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	DynamicPipelineMessageProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	DynamicPipelineMessageProcessor.process(MuleEvent) line: 55	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	ProcessingTimeInterceptor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	ProcessingTimeInterceptor(AbstractEnvelopeInterceptor).processBlocking(MuleEvent) line: 59	
	ProcessingTimeInterceptor(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	AbstractPipeline$ProcessIfPipelineStartedMessageProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	AbstractPipeline$ProcessIfPipelineStartedMessageProcessor(AbstractFilteringMessageProcessor).process(MuleEvent) line: 62	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	AbstractPipeline$1(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	AbstractPipeline$1(AbstractRequestResponseMessageProcessor).processBlocking(MuleEvent) line: 57	
	AbstractPipeline$1(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.doProcess(MuleEvent) line: 50	
	InterceptingChainLifecycleWrapper(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.access$001(InterceptingChainLifecycleWrapper, MuleEvent) line: 22	
	InterceptingChainLifecycleWrapper$1.process(MuleEvent) line: 66	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	InterceptingChainLifecycleWrapper.process(MuleEvent) line: 61	
	Flow$1.process() line: 120	
	Flow$1.process() line: 115	
	ExecuteCallbackInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 16	
	CommitTransactionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 35	
	CommitTransactionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 22	
	HandleExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 30	
	HandleExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 14	
	BeginAndResolveTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 67	
	SuspendXaTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 50	
	RethrowExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 28	
	RethrowExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 13	
	ErrorHandlingExecutionTemplate.execute(ExecutionCallback<MuleEvent>) line: 60	
	ErrorHandlingExecutionTemplate.execute(ExecutionCallback) line: 30	
	Flow.process(MuleEvent) line: 114	
	FlowRefFactoryBean$2.process(MuleEvent) line: 264	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	TransactionalInterceptingMessageProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	TransactionalInterceptingMessageProcessor$1.process() line: 54	
	TransactionalInterceptingMessageProcessor$1.process() line: 51	
	ExecuteCallbackInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 16	
	CommitTransactionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 35	
	CommitTransactionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 22	
	HandleExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 30	
	HandleExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 14	
	BeginAndResolveTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 67	
	ResolvePreviousTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 44	
	SuspendXaTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 50	
	ValidateTransactionalStateInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 40	
	IsolateCurrentTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 41	
	ExternalTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 48	
	RethrowExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 28	
	RethrowExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 13	
	TransactionalErrorHandlingExecutionTemplate.execute(ExecutionCallback<MuleEvent>) line: 110	
	TransactionalErrorHandlingExecutionTemplate.execute(ExecutionCallback) line: 30	
	TransactionalInterceptingMessageProcessor.process(MuleEvent) line: 60	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapperPathSkip(InterceptingChainLifecycleWrapper).doProcess(MuleEvent) line: 50	
	InterceptingChainLifecycleWrapperPathSkip(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.access$001(InterceptingChainLifecycleWrapper, MuleEvent) line: 22	
	InterceptingChainLifecycleWrapper$1.process(MuleEvent) line: 66	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	InterceptingChainLifecycleWrapperPathSkip(InterceptingChainLifecycleWrapper).process(MuleEvent) line: 61	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	DynamicPipelineMessageProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	DynamicPipelineMessageProcessor.process(MuleEvent) line: 55	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	ProcessingTimeInterceptor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	ProcessingTimeInterceptor(AbstractEnvelopeInterceptor).processBlocking(MuleEvent) line: 59	
	ProcessingTimeInterceptor(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	AbstractPipeline$ProcessIfPipelineStartedMessageProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	AbstractPipeline$ProcessIfPipelineStartedMessageProcessor(AbstractFilteringMessageProcessor).process(MuleEvent) line: 62	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	AbstractPipeline$1(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	AbstractPipeline$1(AbstractRequestResponseMessageProcessor).processBlocking(MuleEvent) line: 57	
	AbstractPipeline$1(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.doProcess(MuleEvent) line: 50	
	InterceptingChainLifecycleWrapper(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.access$001(InterceptingChainLifecycleWrapper, MuleEvent) line: 22	
	InterceptingChainLifecycleWrapper$1.process(MuleEvent) line: 66	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	InterceptingChainLifecycleWrapper.process(MuleEvent) line: 61	
	Flow$1.process() line: 120	
	Flow$1.process() line: 115	
	ExecuteCallbackInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 16	
	CommitTransactionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 35	
	CommitTransactionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 22	
	HandleExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 30	
	HandleExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 14	
	BeginAndResolveTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 67	
	SuspendXaTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 50	
	RethrowExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 28	
	RethrowExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 13	
	ErrorHandlingExecutionTemplate.execute(ExecutionCallback<MuleEvent>) line: 60	
	ErrorHandlingExecutionTemplate.execute(ExecutionCallback) line: 30	
	Flow.process(MuleEvent) line: 114	
	FlowRefFactoryBean$2.process(MuleEvent) line: 264	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.doProcess(MuleEvent) line: 50	
	InterceptingChainLifecycleWrapper(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.access$001(InterceptingChainLifecycleWrapper, MuleEvent) line: 22	
	InterceptingChainLifecycleWrapper$1.process(MuleEvent) line: 66	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	InterceptingChainLifecycleWrapper.process(MuleEvent) line: 61	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	DynamicPipelineMessageProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	DynamicPipelineMessageProcessor.process(MuleEvent) line: 55	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	ProcessingTimeInterceptor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	ProcessingTimeInterceptor(AbstractEnvelopeInterceptor).processBlocking(MuleEvent) line: 59	
	ProcessingTimeInterceptor(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	AbstractPipeline$ProcessIfPipelineStartedMessageProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	AbstractPipeline$ProcessIfPipelineStartedMessageProcessor(AbstractFilteringMessageProcessor).process(MuleEvent) line: 62	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	AbstractPipeline$1(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	AbstractPipeline$1(AbstractRequestResponseMessageProcessor).processBlocking(MuleEvent) line: 57	
	AbstractPipeline$1(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.doProcess(MuleEvent) line: 50	
	InterceptingChainLifecycleWrapper(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.access$001(InterceptingChainLifecycleWrapper, MuleEvent) line: 22	
	InterceptingChainLifecycleWrapper$1.process(MuleEvent) line: 66	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	InterceptingChainLifecycleWrapper.process(MuleEvent) line: 61	
	Flow$1.process() line: 120	
	Flow$1.process() line: 115	
	ExecuteCallbackInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 16	
	CommitTransactionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 35	
	CommitTransactionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 22	
	HandleExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 30	
	HandleExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 14	
	BeginAndResolveTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 67	
	SuspendXaTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 50	
	RethrowExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 28	
	RethrowExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 13	
	ErrorHandlingExecutionTemplate.execute(ExecutionCallback<MuleEvent>) line: 60	
	ErrorHandlingExecutionTemplate.execute(ExecutionCallback) line: 30	
	Flow.process(MuleEvent) line: 114	
	Router(AbstractRouter).processBlockingRequest(MuleEvent) line: 108	
	Router(AbstractRouter).processBlocking(MuleEvent) line: 98	
	Router(AbstractRouter).process(MuleEvent) line: 80	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	MulestacLoggingInterceptor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	MulestacLoggingInterceptor(AbstractEnvelopeInterceptor).processBlocking(MuleEvent) line: 59	
	MulestacLoggingInterceptor(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	DynamicPipelineMessageProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	DynamicPipelineMessageProcessor.process(MuleEvent) line: 55	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	ProcessingTimeInterceptor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	ProcessingTimeInterceptor(AbstractEnvelopeInterceptor).processBlocking(MuleEvent) line: 59	
	ProcessingTimeInterceptor(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	AbstractPipeline$ProcessIfPipelineStartedMessageProcessor(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	AbstractPipeline$ProcessIfPipelineStartedMessageProcessor(AbstractFilteringMessageProcessor).process(MuleEvent) line: 62	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 98	
	BlockingProcessorExecutor.execute() line: 59	
	AbstractPipeline$1(AbstractInterceptingMessageProcessorBase).processNext(MuleEvent) line: 99	
	AbstractPipeline$1(AbstractRequestResponseMessageProcessor).processBlocking(MuleEvent) line: 57	
	AbstractPipeline$1(AbstractRequestResponseMessageProcessor).process(MuleEvent) line: 48	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	BlockingProcessorExecutor.executeNext() line: 88	
	BlockingProcessorExecutor.execute() line: 59	
	DefaultMessageProcessorChain.doProcess(MuleEvent) line: 81	
	DefaultMessageProcessorChain(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.doProcess(MuleEvent) line: 50	
	InterceptingChainLifecycleWrapper(AbstractMessageProcessorChain).process(MuleEvent) line: 74	
	InterceptingChainLifecycleWrapper.access$001(InterceptingChainLifecycleWrapper, MuleEvent) line: 22	
	InterceptingChainLifecycleWrapper$1.process(MuleEvent) line: 66	
	ExceptionToMessagingExceptionExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 27	
	MessageProcessorNotificationExecutionInterceptor.execute(MessageProcessor, MuleEvent) line: 111	
	MessageProcessorExecutionTemplate.execute(MessageProcessor, MuleEvent) line: 44	
	InterceptingChainLifecycleWrapper.process(MuleEvent) line: 61	
	AbstractPipeline$3.process(MuleEvent) line: 232	
	HttpMessageProcessorTemplate.routeEvent(MuleEvent) line: 73	
	AsyncResponseFlowProcessingPhase$1.process() line: 73	
	AsyncResponseFlowProcessingPhase$1.process() line: 60	
	ExecuteCallbackInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 16	
	CommitTransactionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 35	
	CommitTransactionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 22	
	HandleExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 30	
	HandleExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 14	
	BeginAndResolveTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 67	
	ResolvePreviousTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 44	
	SuspendXaTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 50	
	ValidateTransactionalStateInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 40	
	IsolateCurrentTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 41	
	ExternalTransactionInterceptor<T>.execute(ExecutionCallback<T>, ExecutionContext) line: 48	
	RethrowExceptionInterceptor.execute(ExecutionCallback<MuleEvent>, ExecutionContext) line: 28	
	RethrowExceptionInterceptor.execute(ExecutionCallback, ExecutionContext) line: 13	
	TransactionalErrorHandlingExecutionTemplate.execute(ExecutionCallback<MuleEvent>) line: 110	
	AsyncResponseFlowProcessingPhase.runPhase(AsyncResponseFlowProcessingPhaseTemplate, MessageProcessContext, PhaseResultNotifier) line: 59	
	AsyncResponseFlowProcessingPhase.runPhase(MessageProcessTemplate, MessageProcessContext, PhaseResultNotifier) line: 36	
	PhaseExecutionEngine$InternalPhaseExecutionEngine.phaseSuccessfully() line: 65	
	PhaseExecutionEngine$InternalPhaseExecutionEngine.phaseSuccessfully() line: 69	
	ThrottlingPhase.runPhase(ThrottlingPhaseTemplate, MessageProcessContext, PhaseResultNotifier) line: 187	
	ThrottlingPhase.runPhase(MessageProcessTemplate, MessageProcessContext, PhaseResultNotifier) line: 58	
	PhaseExecutionEngine$InternalPhaseExecutionEngine.phaseSuccessfully() line: 65	
	GatewayValidationPhase.runPhase(HttpMessageProcessorTemplate, MessageProcessContext, PhaseResultNotifier) line: 93	
	GatewayValidationPhase.runPhase(MessageProcessTemplate, MessageProcessContext, PhaseResultNotifier) line: 49	
	PhaseExecutionEngine$InternalPhaseExecutionEngine.phaseSuccessfully() line: 65	
	CorsPhase.runPhase(HttpMessageProcessorTemplate, MessageProcessContext, PhaseResultNotifier) line: 112	
	CorsPhase.runPhase(MessageProcessTemplate, MessageProcessContext, PhaseResultNotifier) line: 39	
	PhaseExecutionEngine$InternalPhaseExecutionEngine.process() line: 114	
	PhaseExecutionEngine.process(MessageProcessTemplate, MessageProcessContext) line: 41	
	MuleMessageProcessingManager.processMessage(MessageProcessTemplate, MessageProcessContext) line: 32	
	DefaultHttpListener$1.handleRequest(HttpRequestContext, HttpResponseReadyCallback) line: 135	
	GrizzlyRequestDispatcherFilter.handleRead(FilterChainContext) line: 105	
	ExecutorResolver$9.execute(Filter, FilterChainContext) line: 119	
	DefaultFilterChain.executeFilter(FilterExecutor, Filter, FilterChainContext) line: 284	
	DefaultFilterChain.executeChainPart(FilterChainContext, FilterExecutor, int, int, DefaultFilterChain$FiltersState) line: 201	
	DefaultFilterChain.execute(FilterChainContext) line: 133	
	DefaultFilterChain.process(Context) line: 112	
	ProcessorExecutor.execute(Context) line: 77	
	TCPNIOTransport.fireIOEvent(IOEvent, Connection, IOEventLifeCycleListener) line: 539	
	AbstractIOStrategy.fireIOEvent(Connection, IOEvent, IOEventLifeCycleListener, Logger) line: 112	
	ExecutorPerServerAddressIOStrategy.run0(Connection, IOEvent, IOEventLifeCycleListener) line: 126	
	ExecutorPerServerAddressIOStrategy.access$100(Connection, IOEvent, IOEventLifeCycleListener) line: 32	
	ExecutorPerServerAddressIOStrategy$WorkerThreadRunnable.run() line: 149	
	MonitoredThreadPoolExecutor(ThreadPoolExecutor).runWorker(ThreadPoolExecutor$Worker) line: 1149	
	ThreadPoolExecutor$Worker.run() line: 624	
	MonitoredThread(Thread).run() line: 748	

```